### ISSUE TITLE
Chutzpah crashes VS

### ISSUE THREAD

**Issue by tmmueller** [mmanela/chutzpah#322]
Almost every time I rebuild my code, which triggers a test run, VS crashes. I see this stack trace in Event Viewer:

```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ArgumentOutOfRangeException
Stack:
   at System.ThrowHelper.ThrowArgumentOutOfRangeException()
   at System.Collections.Generic.List`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].RemoveAt(Int32)
   at Chutzpah.VS2012.TestAdapter.ChutzpahTestContainerDiscoverer.RemoveTestContainer(Chutzpah.VS2012.TestAdapter.TestFileCandidate)
   at Chutzpah.VS2012.TestAdapter.ChutzpahTestContainerDiscoverer.AddTestContainerIfTestFile(Chutzpah.VS2012.TestAdapter.TestFileCandidate)
   at Chutzpah.VS2012.TestAdapter.ChutzpahTestContainerDiscoverer.OnProjectItemChanged(System.Object, Chutzpah.VS11.EventWatchers.EventArgs.TestFileChangedEventArgs)
   at Chutzpah.VS11.EventWatchers.TestFilesUpdateWatcher.OnChanged(System.Object, System.IO.FileSystemEventArgs)
   at System.IO.FileSystemWatcher.OnChanged(System.IO.FileSystemEventArgs)
   at System.IO.FileSystemWatcher.NotifyFileSystemEventArgs(Int32, System.String)
   at System.IO.FileSystemWatcher.CompletionStatusChanged(UInt32, UInt32, System.Threading.NativeOverlapped*)
   at System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32, UInt32, System.Threading.NativeOverlapped*)
```

I'm running version 3.3.1 in VS 2013. Is there anything I can do to help narrow this down?

**Comment by mmanela** [comment]
Please give me a minimal sample project where you are getting this issue.With that I can test locally and see.

**Comment by tmmueller** [comment]
Hmm, I'm not sure how to do that because I have no idea what is causing the error. I built and ran Chutzpah in debug mode, and of course the error went away. Looking at the code in RemoveTestContainer(), I can't think of a reason it would throw an ArgumentOutOfRangeException unless there were a race condition between getting the index and removing it. Does it run in multiple threads?

**Comment by mmanela** [comment]
It does look like a race condition. I think that could hardened to ensure racing mutations on the cacheContainers object is safe

**Comment by mmanela** [comment]
Fixed in 9265a16
